micro_dl/input/dataset.py

```diff
@@ -171,32 +183,44 @@ def __getitem__(self, index):
         if end_idx >= self.num_samples:
             end_idx = self.num_samples
 
-        input_image = []
-        target_image = []
+        input_fnames = []
+        target_fnames = []
+        aug_ids = []
+        norm_output = self.model_task is not 'segmentation' and self.normalize
+
         aug_idx = 0
         for idx in range(start_idx, end_idx, 1):
-            cur_input_fnames = self.input_fnames.iloc[self.row_idx[idx]]
-            cur_target_fnames = self.target_fnames.iloc[self.row_idx[idx]]
+            input_fnames.append(self.input_fnames.iloc[self.row_idx[idx]])
+            target_fnames.append(self.target_fnames.iloc[self.row_idx[idx]])
             # Select select int randomly that will represent augmentation type
             if self.augmentations:
                 aug_idx = np.random.choice([0, 1, 2, 3, 4, 5], 1)
-            print(aug_idx)
-            cur_input = self._get_volume(cur_input_fnames.split(','),
-                                         aug_idx)
-            cur_target = self._get_volume(cur_target_fnames.split(','),
-                                          aug_idx)
-            # If target is boolean (segmentation masks), convert to float
-            if cur_target.dtype == bool:
-                cur_target = cur_target.astype(np.float64)
-            if self.normalize:
-                cur_input = (cur_input - np.mean(cur_input)) /\
-                             np.std(cur_input)
-                # Only normalize target if we're dealing with regression
-                if self.model_task is not 'segmentation' and self.normalize:
-                    cur_target = (cur_target - np.mean(cur_target)) /\
-                                 np.std(cur_target)
-            input_image.append(cur_input)
-            target_image.append(cur_target)
+            aug_ids.append(aug_idx)
+        order = range(len(aug_ids))
+        with concurrent.futures.ThreadPoolExecutor() as executor:
```

Comment:
wondering how the number of threads is determined here?

Reply:
If you don't specify the number of workers, it will default to the number of processors on the machine you're running on.